cstrml/swork/src/lib.rs

```diff
@@ -824,6 +837,17 @@ impl<T: Config> Module<T> {
         Ok(())
     }
 
+    fn is_report_works_allow(pk: &SworkerPubKey, block_number: u64) -> bool {
+        let mut pk_info = Self::pub_keys(pk);
+        // first time, allow report or reported in the last time
+        let is_ok = pk_info.anchor.is_none() || pk_info.allow_report_slot == block_number || Self::reported_in_slot(pk_info.anchor.clone().unwrap(), block_number - REPORT_SLOT);
+        if !is_ok && block_number > pk_info.allow_report_slot{
+            pk_info.allow_report_slot = block_number + (T::PunishmentDuration::get() as u64 * REPORT_SLOT);
```

Comment:
Since it multi with REPORT_SLOT, maybe named `PunishmentSlots` can be more clear